lib/mayaUsd/ufe/UsdUndoDeleteCommand.cpp

```diff
@@ -37,13 +67,93 @@ UsdUndoDeleteCommand::Ptr UsdUndoDeleteCommand::create(const PXR_NS::UsdPrim& pr
     return std::make_shared<UsdUndoDeleteCommand>(prim);
 }
 
+bool isPrimDeletionAllowed(const PXR_NS::UsdPrim& prim, SdfPrimSpecHandle primSpec)
+{
+
+    auto primStack = prim.GetPrimStack();
```

Comment:
What was the reason to copy this code instead of refactoring [applyCommandRestriction](https://github.com/Autodesk/maya-usd/blob/600c31abcd68ef3fe0047b28fb5f33f6edd0872f/lib/mayaUsd/ufe/private/Utils.cpp#L122) to take the action name as an argument? If we have no good reasons, consider code sharing.

Reply:
The delete command exception did not emit any warnings so there was no feedback when it failed. The path for exceptions was different and i thought it risky to change any behavior there. I can double check it. Point noted on the stage and Ill take care of it.

Reply:
using applyCommandRestriction() from private/Utils will require adding an exception handler catch in TdeleteCmd.cpp
while it works at first glance, im not sure of the ramification to the existing functionality.

<------------add--------------->
    catch (const std::exception& e) {
        wait.stop();
        TDGcommand::setCurrent(nullptr);

        Tstring fmt(rDeleteNonMayaObjectFailed), errMsg;
        Tstring what(e.what());
        errMsg.tFormat(&fmt, &path, &what);
        theCommandEngine->displayError(errMsg);
        return sFailure;
    }
<------------end add--------------->
    catch(...){
		wait.stop();
		TDGcommand::setCurrent( nullptr );
		return sFailure;
	}